test/cmdlineTests/ir_with_assembly_no_memoryguard_creation/output

```diff
@@ -4,7 +4,11 @@ object "D_12" {
     code {
         {
             /// @src 0:82:175  "contract D {..."
+            mstore(64, 128)
             if callvalue() { revert(0, 0) }
+            /// @src 0:115:139  "assembly { mstore(0,0) }"
+            mstore(0, 0)
```

Comment:
This ``mstore`` should still be removed - same goes for a few others and a lot of the ``memoryguard`` ``mstore``s that apparently reappear with the change. I was hoping, we'd not get any such side-effects by the otherwise simple solution in this PR, but seems like we're not so lucky :-).

The reason for this is likely that the redundant store eliminator assumes that all arguments to memory operations are stored in variables at some stage of optimization - we'll need to check how much hassle it is to account for constants in there...

The alternative approach would be to keep the CSE as is and only specifically rematerialize zero literals towards the end of the sequence, but I'd prefer the other approach, unless it complicates the redundant store eliminator too much.

Reply:
Out of my head, I'm not entirely sure *why* this happens - first thing to do is to recreate small example cases of this as ``yulOptimizerTests`` for the redundant store eliminator, trying to create simple cases like
```
{
    mstore(0x40, memoryguard(0x80))
    revert(0,0)
}
```
And check that this will in fact not remove the ``mstore`` - or if we need more complicated cases to trigger that. And compare that to cases in which you have ``let _1 := 0 revert(_1,_1)`` instead - just to see if we can confirm this as cause.

Reply:
Hm, the other two have been resolved, but not this one (with the expression splitter approach).